apps/studio/data/table-editor/table-editor-types.ts

```diff
@@ -79,3 +81,35 @@ export function isMaterializedView(entity?: Entity): entity is MaterializedView
 export function isViewLike(entity?: Entity): entity is View | MaterializedView {
   return isView(entity) || isMaterializedView(entity)
 }
+
+export function postgresTableToEntity(table: PostgresTable): Entity {
+  if (table.columns === undefined || table.relationships === undefined) {
+    throw new Error(
+      'Unable to convert PostgresTable to Entity type. columns and relationships must not be undefined.'
+    )
```

Comment:
should return undefined or something instead of throwing an error? otherwise needs to be handled upstream

Reply:
would these ever be undefined ?
pg meta should always have these defined right, typed to be so

Reply:
_technically_ no for the context of where this function is called (database/tables), although the types coming from `@supabase/postgres-meta` does label `columns` and `relationships` to be optional

Reply:
addressed